src/kernel/crane_constructor/hook_chooser/hook.rs

```diff
@@ -0,0 +1,179 @@
+use crate::kernel::storage::storage::Storage;
+use crate::kernel::storage::storage::Value;
+use crate::Param_to_compare;
+use std::io;
+pub struct Hook{
+    // Vec<Vec<String> Крюки<Характеристики>
+
+    pub hook: Vec<String>,
+    pub bearing: String
+}
+
+
+
+impl Hook{
+
+    pub fn new(_param_comp: Param_to_compare, hooks_storage: &mut Storage) -> Self {
+
+        println!("{}",_param_comp._hook_type);
+        //Выбор крюка
+        let res_hook = Self::hook_select(Self::weight_check(&_param_comp, hooks_storage));
```

Comment:
Try to avoid the calculations in the constructors
Better if your calculations will be performed only when your code started

```rust
/// 
/// Only initializations, 
/// All the parameters stored in the class's fields
let hook = Hook::new(
    params: Params::new(),
    k: K::new(),
    somthing_else_required_for_hook: Some...
);
// 
// All required calculates performing here
result = hook.eval();
```